### DISCUSSION TITLE
Pyrolysis rate with BURN_AWAY and THICKNESS

### DISCUSSION THREAD

**Discussion by IFAB-AS** [firemodels/fds#12568] [Q&A]
We like to ask for clarification for the burning behavior using BURN_AWAY with the following example (HRR and other parameters not given to reduce complexity):
&MESH ID='Mesh01', IJK=10,10,10, XB=0,1,0,1,0,1/   -> 10 cm Mesh
&OBST ID=Block, XB=3.2,3.203,0,0.6,0,0.5, SURF_ID=Block, BULK_DENSITY=2500/
&SURF ID=’Block’, BURN_AWAY=.TRUE., BACKING='VOID', MATL_ID(1,1)='Glass', MATL_MASS_FRACTION(1,1)=1.0, THICKNESS(1)=0.003/
&MATL ID=’Block’, DENSITY=2500.0, REFERENCE_TEMPERATURE=50.0, PYROLYSIS_RATE=10/

OBST-Size = 0.003 x 0.6 x 0.5
Size in SMV/FDS = 0 x 0.6 x 0.5 (THIN OBST, Blocks air flow)
Size for 1d heat transfer = 0.003 x 6x0.003 + 5x0.003 (We assume that each SURF gets the 0.003 thickness and if one cell burns away, the SURF property THICKNESS is applied to new SURFs.)
Size for BULK_DENSITY calculation = OBST-Size

We understand from previous discussions that the best way to simulate the BURN_AWAY is by using the BULK_DENSITY as this calculates the mass from the OBST-line which should be having the real dimensions. In the post https://groups.google.com/g/fds-smv/c/poR-I7uDxj0/m/s_AQor2tFAAJ Randy states that the THICKNESS will nevertheless be used to calculate the pyrolysis rate.
Would this lead to a wrong HRR in the given example? I mean, if we have a fire facing the large side of the block, there would be no or a small difference. But is the setup wrong if we set the fire next to the thin side? Do we also need to specify the THICKNESSES in the larger directions of the block as well (the user guide says “On the SURF line, specify the THICKNESS to be approximately that of a single cell, and also set BACKING=’VOID’.”)
Note that this is only an example and the block can also have other dimensions and not a thin panel.

**Comment by drjfloyd** [comment]
BURN_AWAY was added before we had thin obstrcutions in FDS. So the statement "On the SURF line, specify the THICKNESS to be approximately that of a single cell, and also set BACKING=’VOID’.” is intended for a case where the OBST is multiple grid cells thick. 

In your case use the actual thickness in the thin direction of the OBST.

**Comment by mcgratta** [comment]
If in doubt, create a small test case that includes only the object you seek to burn away. Run the case and confirm that you get the right mass loss. Example cases can be found in the Samples folder under the subfolder called Fires.

**Comment by IFAB-AS** [comment]
I did some tests last year with wood and the mass was correct when BULK_DENSITY is used (mass calculated backwards from the HRR gives the expected value from the volume of OBST with BULK_DENSITY). My question is: Does the THICKNESS setting on the SURF line still influence the burning rate / pyrolysis time as it determines the material temperature / adiabatic surface temperature? 

In the example above, the thickness in one direction needs to be given as 3 mm while for the others the thickness needs to be the grid size (because the THICKNESS will be applied to new created SURFs).

It does not have to be a THIN Obstacle, it can also be 1 x 6 x 5 cells and the same THICKNESS settings (3 mm for the thinner side, 10 cm for the others). Is this the intended way or are we on a completely wrong track?

**Comment by mcgratta** [comment]
When you set `BULK_DENSITY`, FDS calculates the combustible mass of each grid cell that is blocked with an obstruction. As the simulation proceeds, FDS does the usual heat conduction/pyrolysis calculation that it would do if there were no `BURN_AWAY` specified. However, when the mass of a given cell drops to zero due to burning on its multiple sides, it disappears. Of course, there is some inconsistency in all this because the reduced material thickness in the various directions is not going to go to zero at the same time. One face may burn faster than another, and the in-depth calculation is independent. However, `BULK_DENSITY` ensures that at least combustible mass is preserved, which is why we recommend that you use it.

**Comment by IFAB-AS** [comment]
If a THIN OBST is used with BULK_DENSITY, does FDS assume the volume and mass of the OBST is zero? Or are these OBSTs treated with the THICKNESS from the SURF line when THICKEN is not used?

**Comment by mcgratta** [comment]
Depends on what version you use.  I made some corrections to that algorithm so that it uses the XB values on the OBST line. Try giving the thin obstruction an actual thickness and test it on a simple case, like one of the BURN_AWAY cases.

**Comment by IFAB-AS** [comment]
I think it is a good idea to use the XB values, but we see after exporting from Pyrosim two things:
1) Some OBSTS will be thin and have no mass (I assume)
2) Some other OBSTS will be adjusted to the mesh by Pyrosim and the mass is also wrong after this step.

If everything is done manually in FDS (thin obsts do not get BULK_DENSITY etc.), it works. I already have written a ticket to Thunderhead because they might be able to provide a fix that adjusts everything automatically before the export happens. This means:
a) THIN_OBSTS should be treated with THICKNESS from SURF and no BULK_DENSITY parameter is set during export
b) When OBST size is adjusted while export, the BULK_DENSITY for each individual OBST can be adjusted during export accordingly (only for non-thin OBSTS)

For simple cases, this can be done by hand. But we also have imported geometries with 100000 OBSTS and are looking for a better solution for these.
